self-core-impl/src/main/java/com/selfxdsd/core/contracts/ContributorContracts.java

```diff
@@ -175,4 +175,9 @@ public Contract markForRemoval(
     public Iterator<Contract> iterator() {
         return this.contracts.get().iterator();
     }
+
+    @Override
+    public void remove(final Contract contract) {
```

Comment:
@andreoss Similar to ProjectContracts.remove(...), we should first check if the given Contract belongs to the Contributor.

The username and the provider should be equal.

Reply:
@amihaiemil Done.